misk/src/main/kotlin/misk/config/SecretConfigService.kt

```diff
@@ -0,0 +1,25 @@
+package misk.config
+
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.inject.Key
+import misk.DependentService
+import misk.inject.toKey
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Singleton
+class SecretConfigService internal constructor(
+  private val secretConfigLoaderProvider: Provider<SecretConfigLoader>
+) : AbstractIdleService(), DependentService {
+
+  override val consumedKeys = setOf<Key<*>>()
+  override val producedKeys = setOf<Key<*>>(SecretConfigService::class.toKey())
```

Comment:
This is where it gets a little strange for me - config is something I expect to be available during bootstrap, not at some point after the server has already started, and I think there will be a lot of confusion when someone tries to access a secret value prior to this service running (either because they have forgotten to declare a dependency on this service, or because they are in code that is outside of a service)

 I think ideally we'd load the secrets at the time a secret consumer calls `value`; I get that due to the dependency on `ResourceLoader` this has to happen after Guice binding time, but that still allows unfettered access to secrets at any point during server start without the need to interact with the service dependency graph.

Reply:
Okay I think I know where we went sideways. We wanted to be able to decode secrets of arbitrary types: YAMLs, JSONs, etc.

I think we avoid a lot of complexity by saying secrets are only in two forms, YAMLs and Strings? Then we don‘t need the injector to provide us with a configured Moshi.

Well, maybe YAML, Strings, and PEMs. But not anything that requires Injector-layer configuration.